config.go

```diff
@@ -161,85 +160,123 @@ func (config TGFConfig) String() string {
 	return string(bytes)
 }
 
-func (config *TGFConfig) getAwsSession(duration int64) (*session.Session, error) {
-	if cachedSession != nil {
-		return cachedSession, nil
-	}
-	askedForMfa := false
-	options := awsSession.Options{
-		Profile:           config.tgf.AwsProfile,
-		SharedConfigState: awsSession.SharedConfigEnable,
-		AssumeRoleTokenProvider: func() (string, error) {
-			askedForMfa = true
-			fmt.Fprintf(os.Stderr, "Assume Role MFA token code: ")
-			v, err := terminal.ReadPassword(int(os.Stdin.Fd()))
-			fmt.Fprintln(os.Stderr)
-			return string(v), err
-		},
-	}
-	if duration > 0 {
-		options.AssumeRoleDuration = time.Duration(duration) * time.Second
+var cachedAwsConfig *aws.Config
+
+func (tgfConfig *TGFConfig) getAwsConfig(assumeRoleDuration time.Duration) (*aws.Config, error) {
+	if cachedAwsConfig != nil {
+		log.Debug("Using cached AWS config")
+		return cachedAwsConfig, nil
 	}
 
-	session, err := awsSession.NewSessionWithOptions(options)
+	log.Debugf("Creating new AWS config (assumeRoleDuration=%s)", assumeRoleDuration)
+	_config, err := awsConfig.LoadDefaultConfig(
+		context.TODO(),
+		awsConfig.WithSharedConfigProfile(tgfConfig.tgf.AwsProfile),
+		awsConfig.WithAssumeRoleCredentialOptions(func(o *stscreds.AssumeRoleOptions) {
+			o.TokenProvider = stscreds.StdinTokenProvider
+			if assumeRoleDuration > 0 {
+				o.Duration = assumeRoleDuration
+			}
+		}),
+	)
 
-	if err == nil {
-		// We must get the current credentials before verifying the expiration
-		_, err = session.Config.Credentials.Get()
+	if err != nil {
+		return nil, err
 	}
+	config := &_config
```

Comment:
The reason for the funny names is because `_config` is an `aws.Config` while `config` is an `*aws.Config` and you can't really swap the types like that. They need to be separate variables.

I renamed the module because the word `config` was absolutely all over this file and I didn't want any more confusion.